SCRWebService/endpoints/tests/test_summary_care_record.py

```diff
@@ -0,0 +1,59 @@
+"""Tests associated with the Summary Care Record endpoint"""
+import json
+import pathlib
+
+from builder.pystache_message_builder import MessageGenerationError
+from utilities import file_utilities
+from definitions import ROOT_DIR
+from tornado.testing import AsyncHTTPTestCase
+from tornado.web import Application
+from unittest import mock
+from endpoints import summary_care_record
+
+GP_SUMMARY_UPLOAD_URL = "/gp_summary_upload"
+
+complete_data_path = pathlib.Path(ROOT_DIR) / 'endpoints' / 'tests' / 'data' / 'complete_input.json'
+
+
+class TestSummaryCareRecord(AsyncHTTPTestCase):
+    """Tests associated with the Summary Care Record endpoint"""
+    
+    def get_app(self):
+        self.forwarder = mock.MagicMock()
+
+        return Application([
+            (r'/gp_summary_upload', summary_care_record.SummaryCareRecord,
+             dict(forwarder=self.forwarder))
+        ])
+
+    def test_handler_returns_message_processing_response(self):
+        body = file_utilities.FileUtilities.get_file_dict(complete_data_path)
+        self.forwarder.forward_message_to_mhs.return_value = "Nice response message"
+        response = self.fetch(GP_SUMMARY_UPLOAD_URL, method='POST', headers={'interaction-id': '123'},
+                              body=json.dumps(body))
+
+        self.assertEqual(response.body.decode(), "Nice response message")
+
+    def test_empty_body_request_fails(self):
+        body = ''
```

Comment:
id make the body empty json {}, so its still valid. This may be a seperate test though.

Reply:
I think it's correct as it is, since `{}` will parse as valid json it will be left to the template populator to decide if it's missing keys, so from the endpoint perspective it's correct input. With that said, I'll update the test name to reflect that it's a null body vs an empty json request